### DISCUSSION TITLE
Item Pops Out When Selected And Center Item List

### DISCUSSION THREAD

**Discussion by freec3ndy** [LanKuDot/game_modules#37] [Q&A] [answered]
Hi, thank you for your awesome plugin, I have great time using it :)). I'm implementing some specific cases and may need your help.

1. I want my item to disappear when selected and have others to fill in the gap. How can I do that?
2. Is there a way to center box items not by specifying centered element index? Like right in the middle of the list?

Hope I made myself clear. Once again thank you for your hard work!!

**Comment by LanKuDot** [comment, marked as answer]
Hi, thanks for using this asset!
For 1, dynamically changing the number of boxes is not supported. Sorry.
For 2, do you want to center the box by the target box or by something else?

**Reply by freec3ndy** [reply]
Thanks! Let's say, If there are two boxes, I want to center right between those two. And if there are 3, I want to center at the 2nd one.
[1] [2] center-here [3] [4]            
[1] center-here [2]
[1] [2] [3 (center)] [4] [5]           
[1] [2 (center)] [3]
Is it possible?

**Reply by LanKuDot** [reply]
Yeah, if you're using the dragging mode with aligning middle activated.
Add the following code to the [L495 of the `ListPositionCtrl.cs`](https://github.com/LanKuDot/game_modules/blob/810627590219c04e5a122abad0b203f737d5b974/CircularScrollingList/Scripts/ListPositionCtrl.cs#L495)

```csharp
if (_listBoxes.Count % 2 == 0) {
    if (minDeltaDistance > 0)
        minDeltaDistance -= unitPos / 2;
    else
        minDeltaDistance += unitPos / 2;
}
```

It will make the list move the box an half unitPos up or down.

**Comment by feipinxiang** [comment]
How to center boxes when CircularScrollingList.numOfBoxes is greater than BaseListBank.GetContentCount()？
For example, list can display 10 boxes, but bank has only 4 pieces of data. ~~[x][x][x]~~**[1][2][3][4]**~~[x][x][x]~~

**Reply by LanKuDot** [reply]
The box will be centered if the number of the boxes is odd.
If there has even number of boxes, the center point is between the boxes.
